src/game/game.service.ts

```diff
@@ -94,26 +107,61 @@ export class GameService {
     return this.gameRepository.findById(gameId);
   }
 
-  async onRollDice() {
+  onRollDice() {
     const firstDice = Math.ceil(Math.random() * 6);
     const secondDice = Math.ceil(Math.random() * 6);
-    return { firstDice, secondDice };
+    return `${firstDice}:${secondDice}`;
   }
 
-  setRollDiceTimer(
-    game: Partial<GamePayload>,
-    rollDiceCallBack: (game: Partial<GamePayload>) => Promise<void>
+  setTimer(
+    id: string,
+    time: number,
+    args: unknown,
+    callback: (args: unknown) => Promise<void>
   ) {
-    this.clearRollTimer(game.id);
-    const timer = setTimeout(function () {
-      rollDiceCallBack(game);
-    }, game.timeOfTurn);
+    this.clearTimer(id);
+    return new Promise<void>((resolve, reject) => {
+      const timer = setTimeout(async () => {
+        try {
+          await callback(args);
+          resolve();
+        } catch (err) {
+          console.log(err.message);
+          reject(err);
+        }
+      }, time);
+      this.timers.set(id, timer);
 
-    this.rollTimers.set(game.id, timer);
+      this.logger.log(`Timer with id:${id} was set for ${time / 1000} seconds`);
+    });
+  }
 
-    this.logger.log(
-      `User ${game.turnOfUserId} has 30 seconds to roll the dice in game ${game.id}.`
+  calculateEndOfTurn(timeOfTurn: number) {
+    let turnEnds = Date.now();
+    turnEnds += timeOfTurn;
+    return turnEnds.toString();
```

Comment:
why are you transform number to string?

Reply:
In db it is saved as string because we dont want to play with BigInt